.github/workflows/default.yml

```diff
@@ -0,0 +1,146 @@
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+           submodules: 'recursive'
+           fetch-depth: 0
+
+      # Install the host compiler (DMD or LDC)
+      - name: Install ${{ matrix.configuration.compiler }}
+        if: ${{ matrix.configuration.compiler != 'gdc-latest' }}
+        uses: dlang-community/setup-dlang@v1
+        with:
+          compiler: ${{ matrix.configuration.compiler }}
+
+        # GDC not yet supported by setup-dlang
+      - name: Install GDC
+        if: ${{ matrix.configuration.dmd == 'gdmd' }}
+        run: |
+          sudo apt-get install gdc gdmd -y
+          gdc --version
+
+      # Execute all tests via make on POSIX
+      - name: Run make test
+        if: ${{ matrix.host != 'windows-latest' }}
+        env:
+          DC: ${{ matrix.configuration.dmd }}
+        run: make "-j$(nproc)" test
+
+      # Execute all tests via build.bat on Windows
+      - name: Run build.bat test
+        if: ${{ matrix.host == 'windows-latest' }}
+        env:
+          DC: ${{ matrix.configuration.dmd }}
+        run: ./build.bat test
+
+      # Run dub test for all compilers except GDC (which is installed without dub ATM)
+      - name: Run dub test
+        if: ${{ matrix.configuration.dmd != 'gdmd' }}
+        env:
+          DC: ${{ matrix.configuration.dmd }}
+        run: dub test
+
+      # Run linter for exactly one configuration (no need to bother the other one)
+      - name: Run linter
+        if: ${{ matrix.host == 'ubuntu-latest' && matrix.configuration.dmd == 'dmd' }}
+        env:
+          DC: ${{ matrix.configuration.dmd }}
+        run: make lint
+
+      # Parse phobos to check for failures / crashes / ...
+      - name: Checkout Phobos
+        uses: actions/checkout@v2
+        with:
+          repository: dlang/phobos
+          path: phobos
+
+      - name: Apply D-Scanner to Phobos
+        # Takes a very long time / hangs(?) OSX
+        if: ${{ matrix.host != 'macos-latest' || matrix.configuration.dmd != 'dmd' }}
+        env:
+          DC: ${{ matrix.configuration.dmd }}
+        shell: bash
+        run: |
+          # Ensure that D-Scanner is built
+          make
+
+          # Enter Phobos and check that D-Scanner can be found
+          cd phobos
+          ../bin/dscanner --version
+
+          # Allow failures for now, deal with specific errors below
+          set +e
+
+          # Parse each file on it's own
+          for FILE in $(find std -name '*.d');
+          do
+            # echo "$FILE"
+            ../bin/dscanner -S --config=.dscanner.ini "$FILE"
+            EXIT=$?
+            case $EXIT in
+              0)
+                # Success
+                ;;
+
+              1)
+                echo "Warning[$FILE]: Exited with errors"
+                ;;
+
+              137)
+                # Some files seem to require a lot of memory?
+                echo "Warning[$FILE]: Killed due to insufficient memory!"
```

Comment:
This seems like it would be a bug in dscanner, the github actions runners have 7 GB of RAM on Windows/Linux (14 GB of RAM on OSX) so that should never be reached through dscanner.

Reply:
See https://github.com/dlang-community/D-Scanner/pull/853#discussion_r792643324